packages/create-plugin/__e2e__/utils/CreatePlugin.ts

```diff
@@ -102,14 +112,29 @@ export class CreatePlugin {
           `${this.questionsInput[this.currentStep].question}`,
         );
       } else {
-        this.done();
+        this.finishProcess();
       }
     }
 
     return cliExitPromise;
   }
 
-  private done() {
+  public async teardownProcess() {
+    return new Promise<void>((resolve) => {
+      if (this._isProcessExited()) {
+        resolve();
+        return;
+      }
+
+      this.childProcess.on("exit", () => {
+        this.processExited = true;
+        resolve();
+      });
+      this.finishProcess();
+    });
+  }
+
+  private finishProcess() {
     this.childProcess.stdin.end();
   }
```

Comment:
```suggestion
 private finishProcess(): void {
    if (this.childProcess.stdin) {
      this.childProcess.stdin.end();
    }
  }
```
Added a check to ensure stdin exists before calling end(). IMO, this prevents potential runtime errors if stdin is null or undefined.